src/components/MultiSelect.js

```diff
@@ -0,0 +1,51 @@
+import React, { useCallback, useMemo } from 'react'
+import propTypes from '@dhis2/prop-types'
+import { MultiSelectField, MultiSelectOption } from '@dhis2/ui-core'
+
+import { FieldAdapter } from './FieldAdapter.js'
+
+const useChangeHandler = onChange =>
+    useCallback(
+        ({ selected }) => {
+            const value = selected.map(({ value }) => value)
+            onChange({ value }, null)
+        },
+        [onChange]
+    )
+
+const useSelectedOptions = (options, value) =>
+    useMemo(() => {
+        const selectedValues = new Set(value || [])
+        return options.filter(({ value }) => selectedValues.has(value))
+    }, [options, value])
+
+const MultiSelectComponent = ({ options, value, onChange, ...rest }) => {
+    const handleChange = useChangeHandler(onChange)
+    const selectedOption = useSelectedOptions(options, value)
+
+    return (
+        <MultiSelectField
+            {...rest}
+            selected={selectedOption}
+            onChange={handleChange}
+        >
+            {Array.isArray(options)
```

Comment:
I think we could use the same technique here as in the `useSelectedOptions` hook:

```js
const renderOptions = options || []
// ...
return (
  <MultiSelectField ...>
    {renderOptions.map(...)}
  </MultiSelectField>
)
```

That would increase the readability of this component

Reply:
Another good point. I've actually done this slightly differently:
```javascript
// First create the render options, and guarantee this is an array
const renderOptions = Array.isArray(options) ? options : []
// Also use this when getting the selected options so that function won't throw an error
const selectedOptions = useSelectedOptions(renderOptions, value)
```
This also addresses your https://github.com/dhis2/ui-forms/pull/108#discussion_r349973671, which was about the `SingleSelect` but actually applies here too.